mbed/mbed.py

```diff
@@ -2221,6 +2231,23 @@ def compile_(toolchain=None, target=None, profile=False, compile_library=False,
                   + (['-v'] if verbose else [])
                   + args,
                   env=env)
+            
+            if flash:
```

Comment:
Before trying to flash, should you also check that the return code of the compile job indicates success?